core/server/worker/src/test/java/alluxio/worker/block/RegisterStreamerTest.java

```diff
@@ -0,0 +1,364 @@
+              public void onCompleted() {
+                responseObserver.onError(new RuntimeException());
+              }
+            };
+          }
+        });
+
+    RegisterStreamer streamer = createMockRegisterStreamer(mChannel, 1L, 2);
+    assertThrows(InternalException.class, streamer::registerWithMaster);
+  }
+
+  @Test(timeout = 5000)
+  public void testRegisterSuccess() throws Exception {
+    // store workerId to requestCount
+    ConcurrentMap<Long, AtomicInteger> requestCount = new ConcurrentHashMap<>();
+    // create a server that accepts and completes the registration properly
+    createMockService(
+        new BlockMasterWorkerServiceGrpc.BlockMasterWorkerServiceImplBase() {
+          @Override
+          public StreamObserver<RegisterWorkerPRequest> registerWorkerStream(
+              StreamObserver<RegisterWorkerPResponse> responseObserver) {
+            return new NoOpStreamObserver() {
+              @Override
+              public void onNext(RegisterWorkerPRequest value) {
+                long workerId = value.getWorkerId();
+
+                requestCount.putIfAbsent(workerId, new AtomicInteger(0));
+                requestCount.get(workerId).incrementAndGet();
+
+                responseObserver.onNext(RegisterWorkerPResponse.newBuilder().build());
+              }
+
+              @Override
+              public void onError(Throwable t) {
+              }
+
+              @Override
+              public void onCompleted() {
+                responseObserver.onCompleted();
+              }
+            };
+          }
+        });
+
+    // sends 5 requests for one workerId in total
+    long workerId = 1L;
+    RegisterStreamer streamer = createMockRegisterStreamer(mChannel, workerId, 5);
+    streamer.registerWithMaster();
+    assertEquals(1, requestCount.size());
+    assertEquals(5, requestCount.get(workerId).get());
+  }
+
+  @After
+  public void cleanUp() {
+    if (mServer != null && mServer.isServing()) {
+      mServer.shutdown();
+    }
+
+    if (mChannel != null && !mChannel.isShutdown()) {
+      mChannel.shutdown();
+    }
+
+    mServer = null;
+    mChannel = null;
+  }
+
+  // create a mock grpc server that delegates work to mockHandler.
+  // this method modifies mServer and mChannel to the newly created server
+  // and channel
+  private void createMockService(
+      BlockMasterWorkerServiceGrpc.BlockMasterWorkerServiceImplBase mockHandler) throws Exception {
+
+    final BlockMasterWorkerServiceGrpc.BlockMasterWorkerServiceImplBase mockService = mock(
+        BlockMasterWorkerServiceGrpc.BlockMasterWorkerServiceImplBase.class,
+        delegatesTo(mockHandler)
+    );
+
+    mServer = GrpcServerBuilder.forAddress(
+        GrpcServerAddress.create(TEST_ADDRESS),
+        Configuration.global(),
+        ServerUserState.global()
+    )
+        .addService(ServiceType.BLOCK_MASTER_WORKER_SERVICE, new GrpcService(mockService))
+        .build()
+        .start();
```

Comment:
nit: I think our coding style is more like
```suggestion
    mServer = GrpcServerBuilder.forAddress(
          GrpcServerAddress.create(TEST_ADDRESS),
          Configuration.global(),
          ServerUserState.global())
        .addService(ServiceType.BLOCK_MASTER_WORKER_SERVICE, new GrpcService(mockService))
        .build()
        .start();
```

Reply:
Fixed